### DISCUSSION TITLE
Allowing windows to go partially out of bounds

### DISCUSSION THREAD

**Discussion by markus-li** [DustinBrett/daedalOS#120] [Feature Requests]
With bounds set to "main" in rndDefaults windows can't be dragged outside the browser window even partially, to get the feeling of a "normal" desktop that isn't ideal. With it unset windows can be placed outside the browser window, which isn't great either. In short, it would be nice to have windows needing to have a minimal amount of them inside the browser window, but not the whole window. Eventually maybe even with edge and corner resize just like in Windows 10?

**Comment by DustinBrett** [comment]
That is a good point. It can feel a little boxed in as it is now. I will add this to my feature req's. My main issue is making sure Windows don't get lost off screen.

**Comment by markus-li** [comment]
Did a small experiment with this by disabling "bounds" in rndDefaults.ts and replaced setPosition in onDragStop and onResizeStop (useRnd.ts) with this:

```
const enforceBounds = (positionX: number, positionY: number) => {
    let newX = positionX;
    let newY = positionY;
    
    if(positionX + pxToNum(size.width) < 166) {
      newX = 166 - pxToNum(size.width);
    } else if(positionX > viewWidth() - 80) {
      newX = viewWidth() - 80;
    }

    if(positionY < 0) {
      newY = 0;
    } else if(positionY > viewHeight() - 100) {
      newY = viewHeight() - 100;
    }
    
    setPosition({ x: newX, y: newY });
  }
```

This still leaves updating if the browser window is resized. But it's a start.

**Reply by DustinBrett** [reply]
Thanks! I'll check this out when I am working on the Window stuff again. I kinda jump all over the place. Was reducing the bundle size last week.

**Comment by DustinBrett** [comment]
Discussed more in https://github.com/DustinBrett/daedalOS/issues/174

Will prioritize this

**Comment by DustinBrett** [comment]
I just added this in a recent commit. https://github.com/DustinBrett/daedalOS/commit/edd35f5a6b74559e9a1c601dad27b71d99a0f1f3

**Reply by markus-li** [reply]
I'll test it within the next couple of days :) thank you!

**Reply by DustinBrett** [reply]
I will probably improve it eventually to position close as possible to the buffer area. Right now if it's outside the area I've set, it just ignores the drag/drop.
